standardization/src/main/scala/za/co/absa/enceladus/standardization/interpreter/stages/TypeParser.scala

```diff
@@ -211,10 +223,60 @@ object TypeParser {
     override def assemblePrimitiveCastLogic: Column = column.cast(field.dataType)
   }
 
-  private final case class NumericParser(field: StructField,
-                                         path: String,
-                                         origSchema: StructType,
-                                         parent: Option[Parent]) extends ScalarParser
+  private trait NumericParser extends ScalarParser
```

Comment:
Is there a reason to keep NumericParser around?

Reply:
I anticipate it could be useful for patterns in numbers. That's why I kept it still there.